Documentation and Modifications pursuant to PR #35

loggerheads-with-binary opened the pull request:
Hi, I have added documentation and made some modifications as suggested in PR #35  

Additional Commits made:

- Updated Version for Semver
- Modified Format::format to include &impl Map<K,V> and updated documentation
  - Modifed Format::format and its implementation for both std::String and std::str::str to accept &impl Map<K,V> instead of HashMap
  - Added documentation in both Map Trait for auto docs.rs documentation
  - Added documentation in the README.md file

loggerheads-with-binary commented:
I have updated the semver to 0.3.0 

If possible, could you do a `cargo publish`. 
It will update the crate on [crates.io](https://crates.io) and also update the documentation on [docs.rs](https://docs.rs)

vitiral commented on README.md:
```diff
@@ -1,9 +1,26 @@
-# strfmt: rust library for formatting dynamic strings
+# Index  
```
Put the index after the first line + intro and make it `##`

Also rename it "Table of Contents". Otherwise I like it :D

loggerheads-with-binary replied:
Done

vitiral commented on src/maptrait.rs:
```diff
@@ -3,9 +3,14 @@ use std::str::FromStr;
 pub trait Map<K , V> where K: Hash + Eq + FromStr, V: DisplayStr{
     
-    ///Prioritized first 
+    /// Prioritized first 
```
What does this comment mean? Please remove it.

loggerheads-with-binary replied:
There are two functions in the `Map` trait, `get `and `get_owned`. get is prioritized first. Only if `get `returns `None`, the `get_owned `method call will be performed. That's why the comment is there.

vitiral commented on src/maptrait.rs:
```diff
@@ -3,9 +3,14 @@ use std::str::FromStr;
 use std::collections::HashMap; 
 
+
+///Trait for custom implementation of formatting 
```
I think replace this with "Represents a generic Map. Implemented for HashMap and (optionally) serde's Map"

loggerheads-with-binary replied:
Done

vitiral commented on README.md:
```diff
@@ -36,12 +53,53 @@ strfmt does not support empty identifiers (i.e. `{}` or `{:<10}`. Integer
 key == "1")
 
-## Legacy
+## Using the Map trait for customizing the display<a name="map-trait"></a> 
```
This section can be removed. It's in the API docs

loggerheads-with-binary replied:
As a counterpoint, I think the documentation of the Map trait should probably be there in the README since it highlights on user-end customization which is very prominent in a lot of README files. 

However if you still think we should remove this from the documentation, I'll take it off. 

PS: One of the reasons I had pulled your crate into my project as over some of the other ones was because I was easily able to override the HashMap requirement and customize my trait to cover lazy implementations

loggerheads-with-binary commented:
Added optional support for Hashbrown crate as well 
Additionally, added a few tests for checking the hashbrowns implementation and the serde_json implementation 
All tests passing

vitiral commented:
Sorry to reverse on my previous statement, but I'm going back to thinking that none of this belongs here.

If rust had a "standard" Map trait then by all means I would add support for it, but without it this is just bloat with minimal benefits.

In nearly every case I can imagine the caller can simply convert their `SpecialMap` into a `HashMap` and call strfmt.

Since all we seem to need is the `get` method the user could simply pass in a `Fn` type that accomplishes this -- no hash-map needed.

I'm closing this PR and am rolling back the previous one. If you'd like to contribute the support for a function (aka closure) which performs lookups I'd love to review that. If Rust ever adds a `Map` trait then I'd be happy to accept that as well.

vitiral commented:
https://github.com/vitiral/strfmt/issues/17#issuecomment-1721534661 

`strfmt_map` is that function, so it already exists. Let's stabilize that instead.
